### ISSUE TITLE
Rails 6 Multiple databases: model connection_config missing database property

### ISSUE THREAD

**Issue by littletime** [rails/rails#38440]
Using Rails 6, and setting up a multiple databases configuration.
I want to separate some fairly static data from the rest of the app, so I have the primary (app) and products (static) databases.
Migrations work fine, both dbs are created with the appropriate schemas. Problem is the models that are connected to the `products` db cannot connect.

Here is my database.yml file:

```
default: &default
  adapter: postgresql
  encoding: unicode
  host: db
  username: postgres
  password:
  pool: 5

products: &products
  <<: *default
  migrations_paths: db/products_migrate

development:
  primary:
    <<: *default
    database: myapp_development
  products:
    <<: *products
    database: products_development
```

And here are my models:
product_record.rb
```
class ProductRecord < ApplicationRecord
  self.abstract_class = true

  connects_to database: { writing: :products, reading: :products }
end
```
product.rb
```
class Product < ProductRecord
end
```

This seems fine from what I see in the docs. But in Rails console, `Product.all` tells me 
```
ActiveRecord::StatementInvalid (PG::UndefinedTable: ERROR:  relation "products" does not exist)
LINE 1: SELECT "products".* FROM "products" LIMIT $1
```

So, in the console, I check my model's connection_config:
```
irb(main):009:0> Product.connection_config
=> {:adapter=>"postgresql", :encoding=>"unicode", :host=>"db", :username=>"postgres", :password=>nil, :pool=>5, :migrations_paths=>"db/products_migrate"}
```

So there is no `database` property. Really weird thing is, it does get the "migration_paths" property, so the configuration in the model works, but it just omits the `database` property, resulting in rails not connecting to any db.

### System configuration
**Rails version**:
6.0.2.1
**Ruby version**:
2.5

**Comment by eileencodes** [comment]
Hey @littletime thanks for the report. Can you make a sample application that reproduces your issue? It's hard to debug what's wrong without that.

At a cursory glance there's a couple things that are off.

1) There is a known bug that the yaml alias cannot be the same as one of the entries in the 3-tier config. So you can't have a `products` alias and a `products` database. Can you try changing that and see if it fixes anything? I would have expected this to blow up on `db:create` though. 
2) `connects_to database: { writing: :products, reading: :products }` can work but it's kind of useless since you don't have a replica connection. Without a replica connection adding `reading` isn't super useful. You can do either `connects_to database: { writing: :products }` or `establish_connection :products` and it will default to a writing connection.

**Comment by littletime** [comment]
Hey @eileencodes,

thanks for your answer. I applied your advices: changed the alias name so it wouldn't be the same as the db configuration, and only let the `connects_to database: { writing: :products }` in my model.

At first, it was still behaving the same way, so I deleted all the databases that were created previously, re-ran the `db:create` and `db:migrate` with the new `database.yml`, and it worked!

So I guess it definitely had something to do with the alias being the same as the configuration.

Thank you very much for your help! :)
